### DISCUSSION TITLE
undefined reference to `HAL_LTDC_ConfigLayer'

### DISCUSSION THREAD

**Discussion by epikao** [stm32duino/Arduino_Core_STM32#1532] [Libraries] [answered]
Hello, 
I use platformio with STM32F746 disco board and arduino framework. I now wanted to activate the display interface.
I added LTDC.h and LTDC_F746_Discovery.h in the main folder and the voltmeter example in the main file + 3 includes:
#include <SPI.h>
#include <Wire.h>
#include <Adafruit_I2CDevice.h>

Then I activated the HAL LTDC in the variant_DISCO_F746NG.h file.

I receive one compiler error, see also below. What's wrong here?

```
c:/users/---/.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld.exe: .pio/build/disco_f746ng/src/main.cpp.o: in function setup': main.cpp:(.text.setup+0x98): undefined reference to HAL_LTDC_Init'
c:/users/---/.platformio/packages/toolchain-gccarmnoneeabi/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld.exe: main.cpp:(.text.setup+0xcc): undefined reference to `HAL_LTDC_ConfigLayer'
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\disco_f746ng\firmware.elf] Error 1
```

**Comment by fpistm** [comment, marked as answer]
Hi @epikao 
yes you had to enable the HAL LTDC module.
About your issue I don't know as I do not use PIO and don't know how this is handled maybe a cache issue.

**Reply by epikao** [reply]
> Hi @epikao
> yes you had to enable the HAL LTDC module.
> About your issue I don't know as I do not use PIO and don't know how this is handled maybe a cache issue.

the error is `undefined reference to HAL_LTDC_ConfigLayer ` in the setup function... could it be that I miss an include in the main file or a define in a h file - or something in the setup function?

**Reply by fpistm** [reply]
I don't think. Seems more than HAL LTDC is not correctly enabled and so the .o of the hal ltcd source file does not contain it.
Ensure you have a clean project to avoid any cache issue.
As said I don't know PIO build configuration so hard to tell what goes wrong.

**Comment by epikao** [comment]
Thank you very much, it was indeed the cache. "Clean all" button did the job.
